src/components/input/bl-input.ts

```diff
@@ -284,6 +294,7 @@ export default class BlInput extends FormControlMixin(LitElement) {
 
   firstUpdated() {
     this.setValue(this.value);
+    if (!this.icon) this.icon = inputTypeIcons[this.type];
```

Comment:
I think it's better to have a test for this.

Also some examples in storybook would be helpful to check them visually.